app/controllers/commodities_controller.rb

```diff
@@ -26,6 +25,17 @@ def commodities_by_code
   def fetch_commodities
     @commodities ||= {}
 
+    if is_heading_id?
+      heading_id = params[:id].slice(0, 4)
+      heading = HeadingPresenter.new(Heading.find(heading_id, query_params))
+
+      if heading.declarable? # stay in commodity page
+        @commodities[:uk] = heading
+
+        return
+      end
+    end
+
```

Comment:
This is like a guard_clause, I am thinking to move this to a different `before_action`